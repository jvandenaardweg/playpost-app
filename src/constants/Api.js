export const PRODUCTION_API_URL = 'http://medium-audio.herokuapp.com'
export const DEVELOPMENT_API_URL = 'http://localhost:3000'
