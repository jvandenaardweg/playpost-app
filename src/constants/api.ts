const PRODUCTION_API_URL = 'https://api.postplay.app';
const STAGING_API_URL = 'https://api-staging.postplay.app';
const DEVELOPMENT_API_URL = 'http://localhost:3000';

interface ApiEnvironments {
  [key: string]: string;
}

const environments: ApiEnvironments = {
  production: PRODUCTION_API_URL,
  staging: STAGING_API_URL,
  development: DEVELOPMENT_API_URL
};

export const API_URL = environments[process.env.NODE_ENV || 'production'];
