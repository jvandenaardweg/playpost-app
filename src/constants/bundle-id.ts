export const APP_BUNDLE_ID = 'com.aardwegmedia.playpost';
