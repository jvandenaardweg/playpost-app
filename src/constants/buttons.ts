export const defaultHitslop = {
  top: 20,
  right: 20,
  bottom: 20,
  left: 20
};

export const smallHitslop = {
  top: 5,
  right: 5,
  bottom: 5,
  left: 5
};

export const mediumHitslop = {
  top: 10,
  right: 10,
  bottom: 10,
  left: 10
};
