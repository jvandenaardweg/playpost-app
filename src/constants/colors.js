const tintColor = '#037DE2';
const black = '#000000';
const grayDark = '#333C46';
const grayLight = '#cccccc';
const grayLightest = '#EEF0F4';
const appBackground = '#EBEBEB';
const white = '#ffffff';
const blue = tintColor;
const red = 'red';
const orange = '#EAEB5E';
const orangeDark = '#666804';

export default {
  tintColor,
  black,
  grayDark,
  grayLight,
  grayLightest,
  white,
  blue,
  red,
  orange,
  appBackground,

  tabIconDefault: grayLight,
  tabIconSelected: tintColor,
  tabBar: white,

  articleBackground: white,

  titleDefault: black,
  paragraphDefault: black,
  paragraphGrayed: grayDark,
  borderDefault: grayLightest,

  controlButtonDefault: black,
  controlButtonActive: tintColor,

  errorBackground: red,
  errorText: white,
  warningBackground: orange,
  warningText: orangeDark,
  noticeBackground: tintColor,
  noticeText: white,
};
