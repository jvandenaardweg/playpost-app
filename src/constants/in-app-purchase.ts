import { Platform } from 'react-native';

export const SUBSCRIPTION_PRODUCT_ID_FREE = Platform.select({
  ios: 'free',
  android: 'free'
});

export const SUBSCRIPTION_PRODUCT_ID_PREMIUM = Platform.select({
  ios: 'com.aardwegmedia.playpost.premium',
  android: 'com.aardwegmedia.playpost.android.premium'
});

export const SUBSCRIPTION_PRODUCT_ID_PLUS = Platform.select({
  ios: 'com.aardwegmedia.playpost.subscription.plus',
  android: 'com.aardwegmedia.playpost.android.plus'
});

export const SUBSCRIPTION_PRODUCT_IDS = [SUBSCRIPTION_PRODUCT_ID_PREMIUM, SUBSCRIPTION_PRODUCT_ID_PLUS]
