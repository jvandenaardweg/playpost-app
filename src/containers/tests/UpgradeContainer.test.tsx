import React from 'react';
// import { render, RenderAPI, fireEvent } from 'react-native-testing-library';
import renderer from 'react-test-renderer';

import { Props, UpgradeContainerComponent } from '../UpgradeContainer';

jest.mock('../../navigation/NavigationService');

import mockApplePurchases from '../../../tests/__mocks__/apple-purchases';
import mockInAppSubscriptionApple from '../../../tests/__mocks__/in-app-subscription-apple';
import mockInAppSubscriptionGoogle from '../../../tests/__mocks__/in-app-subscription-google';
import mockSubscriptions from '../../../tests/__mocks__/subscriptions';

import { SUBSCRIPTION_PRODUCT_ID_FREE, SUBSCRIPTION_PRODUCT_ID_PREMIUM } from '../../constants/in-app-purchase';
import { ALERT_SUBSCRIPTION_RESTORE_PLATFORM_ANDROID, ALERT_SUBSCRIPTION_RESTORE_PLATFORM_IOS, ALERT_SUBSCRIPTION_UPGRADE_PLATFORM_ANDROID, ALERT_SUBSCRIPTION_UPGRADE_PLATFORM_IOS, ALERT_TITLE_ERROR } from '../../constants/messages';

const getInAppSubscriptionsHandler = jest.fn();
const getUserHandler = jest.fn();
const navigateHandler = jest.fn();
const navigationGetParamHandler = jest.fn();
const navigationGoBackHandler = jest.fn();
const setIsLoadingUpgradeHandler = jest.fn();
const setIsLoadingRestoreHandler = jest.fn();
const setLocalPurchaseHistoryHandler = jest.fn();

const defaultProps: Props = {
  subscriptionsError: '',
  validationResult: null,
  activeSubscriptionProductId: '',
  centeredSubscriptionProductId: '',
  userDetails: null,
  totalAvailableVoices: 10,
  isSubscribed: false,
  userIsEligibleForTrial: false,
  isLoadingUpgrade: false,
  isLoadingRestore: false,
  activeInAppSubscription: null,
  totalAvailableUnsubscribedVoices: 28,
  availableInAppSubscriptions: [],
  getInAppSubscriptions: getInAppSubscriptionsHandler,
  getUser: getUserHandler,
  setIsLoadingUpgrade: setIsLoadingUpgradeHandler,
  setIsLoadingRestore: setIsLoadingRestoreHandler,
  setLocalPurchaseHistory: setLocalPurchaseHistoryHandler,

  navigation: {
    navigate: navigateHandler,
    getParam: navigationGetParamHandler,
    goBack: navigationGoBackHandler,
  } as any
};

describe('UpgradeContainerComponent', () => {

  describe('rendering', () => {
    let wrapper: renderer.ReactTestRenderer;

    beforeEach(() => {
      const props = {
        ...defaultProps
      }

      wrapper = renderer.create(<UpgradeContainerComponent {...props} />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });

  describe('interacting', () => {
    let wrapper: renderer.ReactTestRenderer;

    afterEach(() => {
      jest.resetAllMocks();
      jest.resetModules();
    });

    beforeEach(() => {
      const props = {
        ...defaultProps
      }

      wrapper = renderer.create(<UpgradeContainerComponent {...props} />);
    });

    it('handleOnPressUpgrade() should correctly handle an downgrade to free', async () => {
      const testProductId = SUBSCRIPTION_PRODUCT_ID_FREE;
      const testInstance = wrapper.root.instance;

      const spySubscriptionAlert = jest.spyOn(testInstance, 'showManageSubscriptionAlert');

      // User is not downgrading from a paid subscription
      testInstance.isDowngradePaidSubscription = jest.fn().mockReturnValue(false)

      // User is downgrading to a free subscription
      testInstance.isDowngradeFreeSubscription = jest.fn().mockReturnValue(true)

      await testInstance.handleOnPressUpgrade(testProductId);

      expect(spySubscriptionAlert).toHaveBeenCalledTimes(1);
      expect(spySubscriptionAlert).toHaveBeenCalledWith(
        'Downgrade to Free?',
        'To downgrade to Free you need to cancel your current subscription. Cancelling a subscription can only be done through iTunes.\n\n Press \"Manage Subscriptions\" below to manage your subscriptions.');
    });

    it('handleOnPressUpgrade() should correctly handle an downgrade from a higher subscription', async () => {
      const testProductId = SUBSCRIPTION_PRODUCT_ID_PREMIUM;
      const testInstance = wrapper.root.instance;

      const spySubscriptionAlert = jest.spyOn(testInstance, 'showManageSubscriptionAlert');

      // User is downgrading from a paid subscription to free
      testInstance.isDowngradePaidSubscription = jest.fn().mockReturnValue(true)

      // User is not downgrading to a free subscription
      testInstance.isDowngradeFreeSubscription = jest.fn().mockReturnValue(false)

      await testInstance.handleOnPressUpgrade(testProductId);

      expect(spySubscriptionAlert).toHaveBeenCalledTimes(1);
      expect(spySubscriptionAlert).toHaveBeenCalledWith(
        'Downgrading Subscription?',
        'Downgrading a subscription can only be done through iTunes.\n\n Press "Manage Subscriptions" below to manage your subscriptions.');
    });

    it('handleOnPressUpgrade() should correctly handle an upgrade', async () => {
      const testProductId = SUBSCRIPTION_PRODUCT_ID_PREMIUM;

      const testInstance = wrapper.root.instance;

      const spySubscriptionAlert = jest.spyOn(testInstance, 'showManageSubscriptionAlert');
      const spyRequestSubscription = jest.spyOn(testInstance, 'requestSubscription');
      const spySetIsLoadingUpgrade = jest.spyOn(testInstance.props, 'setIsLoadingUpgrade');

      // User is not downgrading from a paid subscription
      testInstance.isDowngradePaidSubscription = jest.fn().mockReturnValue(false)

      // User is not downgrading to a free subscription
      testInstance.isDowngradeFreeSubscription = jest.fn().mockReturnValue(false)

      // Test an upgrade to a higher subscription level
      await testInstance.handleOnPressUpgrade(testProductId);

      expect(spySubscriptionAlert).toHaveBeenCalledTimes(0);
      expect(spyRequestSubscription).toHaveBeenCalledTimes(1);
      expect(spySetIsLoadingUpgrade).toHaveBeenCalledTimes(1);
      expect(spySetIsLoadingUpgrade).toHaveBeenCalledWith(true);
      expect(spyRequestSubscription).toHaveBeenCalledWith(testProductId, '', '');
      expect(testInstance.state.centeredSubscriptionProductId).toBe(testProductId);

    });

    it('handleOnPressUpgrade() should correctly handle an error on an upgrade', async () => {
      const testProductId = SUBSCRIPTION_PRODUCT_ID_PREMIUM;

      const testInstance = wrapper.root.instance;

      const spyRequestSubscription = jest.spyOn(testInstance, 'requestSubscription').mockRejectedValueOnce(new Error('Some error!'));
      const spyShowErrorAlert = jest.spyOn(testInstance, 'showErrorAlert');
      const spySetIsLoadingUpgrade = jest.spyOn(testInstance.props, 'setIsLoadingUpgrade');

      // User is not downgrading from a paid subscription
      testInstance.isDowngradePaidSubscription = jest.fn().mockReturnValue(false)

      // User is not downgrading to a free subscription
      testInstance.isDowngradeFreeSubscription = jest.fn().mockReturnValue(false)

      // Test an upgrade to a higher subscription level
      await testInstance.handleOnPressUpgrade(testProductId);

      expect(spyRequestSubscription).toHaveBeenCalledTimes(1);
      expect(spySetIsLoadingUpgrade).toHaveBeenCalledTimes(2);
      expect(spySetIsLoadingUpgrade).toHaveBeenLastCalledWith(false);
      expect(spyRequestSubscription).toHaveBeenCalledWith(testProductId, '', '');
      expect(spyShowErrorAlert).toHaveBeenCalledTimes(0); // The error is handled in SubscriptionHandlerContainer

    });

    it('handleOnPressUpgrade() should prevent the user from upgrading if the user is on iOS but has an Google Play subscription active', async () => {
      const Platform = require('react-native').Platform;
      Platform.OS = 'ios';

      const props = {
        ...defaultProps,
        activeInAppSubscription: mockInAppSubscriptionGoogle
      }

      wrapper.update(<UpgradeContainerComponent {...props} />)

      const testInstance = wrapper.root.instance;

      const spyShowErrorAlert = jest.spyOn(testInstance, 'showErrorAlert').mockReturnValueOnce('');

      const testProductId = SUBSCRIPTION_PRODUCT_ID_PREMIUM;

      const spyRequestSubscription = jest.spyOn(testInstance, 'requestSubscription');

      // Test an upgrade to a higher subscription level
      await testInstance.handleOnPressUpgrade(testProductId);

      expect(spyRequestSubscription).toHaveBeenCalledTimes(0);
      expect(spyShowErrorAlert).toHaveBeenCalledTimes(1);
      expect(spyShowErrorAlert).toHaveBeenCalledWith(ALERT_TITLE_ERROR, ALERT_SUBSCRIPTION_UPGRADE_PLATFORM_ANDROID);

    });

    it('handleOnPressUpgrade() should prevent the user from upgrading if the user is on Android but has an Apple App Store subscription active', async () => {
      const Platform = require('react-native').Platform;
      Platform.OS = 'android';

      const props = {
        ...defaultProps,
        activeInAppSubscription: mockInAppSubscriptionApple
      }

      wrapper.update(<UpgradeContainerComponent {...props} />)

      const testInstance = wrapper.root.instance;

      const spyShowErrorAlert = jest.spyOn(testInstance, 'showErrorAlert').mockReturnValueOnce('');

      const testProductId = SUBSCRIPTION_PRODUCT_ID_PREMIUM;

      const spyRequestSubscription = jest.spyOn(testInstance, 'requestSubscription');

      // Test an upgrade to a higher subscription level
      await testInstance.handleOnPressUpgrade(testProductId);

      expect(spyRequestSubscription).toHaveBeenCalledTimes(0);
      expect(spyShowErrorAlert).toHaveBeenCalledTimes(1);
      expect(spyShowErrorAlert).toHaveBeenCalledWith(ALERT_TITLE_ERROR, ALERT_SUBSCRIPTION_UPGRADE_PLATFORM_IOS);

    });

    it('getLatestPurchase() should return the latest purchase from the array', () => {
      const testInstance = wrapper.root.instance;

      expect(testInstance.getLatestPurchase(mockApplePurchases)).toMatchObject(mockApplePurchases[0]);
    })

    it('isDowngradeFreeSubscription() should return true/false', () => {
      const props = {
        ...defaultProps,
        activeSubscriptionProductId: SUBSCRIPTION_PRODUCT_ID_FREE,
        subscriptions: mockSubscriptions
      }

      wrapper.update(<UpgradeContainerComponent {...props} />)
      const testInstance = wrapper.root.instance;

      expect(testInstance.isDowngradeFreeSubscription(SUBSCRIPTION_PRODUCT_ID_FREE)).toBe(false)
      expect(testInstance.isDowngradeFreeSubscription(SUBSCRIPTION_PRODUCT_ID_PREMIUM)).toBe(false)
      expect(testInstance.isDowngradeFreeSubscription('com.aardwegmedia.playpost.subscriptions.plus')).toBe(false)
    })

    it('isDowngradePaidSubscription() should return true/false', () => {
      const props = {
        ...defaultProps,
        activeSubscriptionProductId: 'com.aardwegmedia.playpost.subscriptions.plus',
        subscriptions: mockSubscriptions
      }

      wrapper.update(<UpgradeContainerComponent {...props} />)
      const testInstance = wrapper.root.instance;

      expect(testInstance.isDowngradePaidSubscription(SUBSCRIPTION_PRODUCT_ID_FREE)).toBe(false)
      expect(testInstance.isDowngradePaidSubscription(SUBSCRIPTION_PRODUCT_ID_PREMIUM)).toBe(false)
    })

    it('handleOnPressRestore() should correctly handle when a user has previous purchases', async () => {
      const testInstance = wrapper.root.instance;

      const spyGetAvailablePurchases = jest.spyOn(testInstance, 'getAvailablePurchases').mockReturnValueOnce(mockApplePurchases)
      const spyGetLatestPurchase = jest.spyOn(testInstance, 'getLatestPurchase').mockReturnValueOnce(mockApplePurchases[0]);
      const spyShowErrorAlert = jest.spyOn(testInstance, 'showErrorAlert');
      const spySetIsLoadingRestore = jest.spyOn(testInstance.props, 'setIsLoadingRestore');
      const spyGetUser = jest.spyOn(testInstance.props, 'getUser');
      // const spyValidateSubscriptionReceipt = jest.spyOn(testInstance.props, 'validateSubscriptionReceipt');


      // Simulate a click on "Restore previous purchase"
      await testInstance.handleOnPressRestore();

      expect(spySetIsLoadingRestore).toHaveBeenCalledTimes(2);
      expect(spySetIsLoadingRestore).toHaveBeenNthCalledWith(1, true);
      expect(spySetIsLoadingRestore).toHaveBeenNthCalledWith(2, false);

      expect(spyGetAvailablePurchases).toHaveBeenCalledTimes(1);
      expect(spyGetAvailablePurchases).toHaveReturnedWith(mockApplePurchases)

      expect(spyShowErrorAlert).toHaveBeenCalledTimes(0);

      expect(spyGetUser).toHaveBeenCalledTimes(1);

      expect(spyGetLatestPurchase).toHaveBeenCalledTimes(1);
      expect(spyGetLatestPurchase).toHaveReturnedWith(mockApplePurchases[0]);

      // expect(spyValidateSubscriptionReceipt).toHaveBeenCalledTimes(1);
      // expect(spyValidateSubscriptionReceipt).toHaveBeenCalledWith(mockApplePurchases[0].productId, mockApplePurchases[0].transactionReceipt, 'ios');
    });

    it('handleOnPressRestore() should correctly handle when a user has no previous purchases', async () => {
      const testInstance = wrapper.root.instance;

      const spyGetAvailablePurchases = jest.spyOn(testInstance, 'getAvailablePurchases').mockReturnValueOnce([]);
      const spyShowErrorAlert = jest.spyOn(testInstance, 'showErrorAlert');
      const spySetIsLoadingRestore = jest.spyOn(testInstance.props, 'setIsLoadingRestore');

      // Simulate a click on "Restore previous purchase"
      await testInstance.handleOnPressRestore();

      expect(spySetIsLoadingRestore).toHaveBeenCalledTimes(2);
      expect(spySetIsLoadingRestore).toHaveBeenLastCalledWith(false);

      expect(spyGetAvailablePurchases).toHaveBeenCalledTimes(1);
      expect(spyGetAvailablePurchases).toHaveReturnedWith([])

      expect(spyShowErrorAlert).toHaveBeenCalledTimes(1);
      expect(spyShowErrorAlert).toHaveBeenCalledWith('Nothing to restore', `We could not find any previous purchase to restore. If you think this is incorrect, please contact our support.`);

    });

    it('handleOnPressRestore() should show an error alert when an error throws inside', async () => {
      const testInstance = wrapper.root.instance;

      jest.spyOn(testInstance, 'getAvailablePurchases').mockRejectedValueOnce(new Error('Some error!'));
      const spyShowErrorAlert = jest.spyOn(testInstance, 'showErrorAlert');
      const spySetIsLoadingRestore = jest.spyOn(testInstance.props, 'setIsLoadingRestore');

      // Simulate a click on "Restore previous purchase"
      await testInstance.handleOnPressRestore();

      expect(spySetIsLoadingRestore).toHaveBeenCalledTimes(2);
      expect(spySetIsLoadingRestore).toHaveBeenLastCalledWith(false);

      expect(spyShowErrorAlert).toHaveBeenCalledTimes(1);
      expect(spyShowErrorAlert).toHaveBeenCalledWith('Restore purchase error', `Some error!`);

    });

    it('handleOnPressRestore() should show an error alert when the active subscription is from the Google Play Store, but the platform is iOS', async () => {
      const Platform = require('react-native').Platform;
      Platform.OS = 'ios';

      const props = {
        ...defaultProps,
        activeInAppSubscription: mockInAppSubscriptionGoogle
      }

      wrapper.update(<UpgradeContainerComponent {...props} />)

      const testInstance = wrapper.root.instance;

      const spyShowErrorAlert = jest.spyOn(testInstance, 'showErrorAlert').mockReturnValueOnce('');

      // Simulate a click on "Restore previous purchase"
      await testInstance.handleOnPressRestore();

      expect(spyShowErrorAlert).toHaveBeenCalledTimes(1);
      expect(spyShowErrorAlert).toHaveBeenCalledWith(ALERT_TITLE_ERROR, ALERT_SUBSCRIPTION_RESTORE_PLATFORM_IOS);

    });

    it('handleOnPressRestore() should show an error alert when the active subscription is from the Apple App Store, but the platform is Android', async () => {
      const Platform = require('react-native').Platform;
      Platform.OS = 'android';

      const props = {
        ...defaultProps,
        activeInAppSubscription: mockInAppSubscriptionApple
      }

      wrapper.update(<UpgradeContainerComponent {...props} />)

      const testInstance = wrapper.root.instance;

      const spyShowErrorAlert = jest.spyOn(testInstance, 'showErrorAlert').mockReturnValueOnce('');

      // Simulate a click on "Restore previous purchase"
      await testInstance.handleOnPressRestore();

      expect(spyShowErrorAlert).toHaveBeenCalledTimes(1);
      expect(spyShowErrorAlert).toHaveBeenCalledWith(ALERT_TITLE_ERROR, ALERT_SUBSCRIPTION_RESTORE_PLATFORM_ANDROID);

    });
  });
});
