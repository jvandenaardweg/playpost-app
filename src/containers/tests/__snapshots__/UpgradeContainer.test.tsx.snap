// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpgradeContainerComponent rendering should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "backgroundColor": "#EEEEEE",
        "paddingTop": 12,
      }
    }
  >
    <View>
      <RCTScrollView
        decelerationRate={0}
        horizontal={true}
        pagingEnabled={true}
        scrollEnabled={true}
        showsHorizontalScrollIndicator={false}
        snapToInterval={696}
        style={
          Object {
            "backgroundColor": "#EEEEEE",
            "marginBottom": 12,
            "paddingLeft": 0,
            "paddingRight": 0,
          }
        }
      >
        <View />
      </RCTScrollView>
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderTopColor": "#e5e5e5",
            "borderTopWidth": 1,
            "padding": 20,
          }
        }
      >
        <View
          style={
            Object {
              "marginBottom": 12,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#040404",
                  "fontFamily": "Inter-SemiBold",
                  "fontSize": 28,
                  "lineHeight": 34,
                },
                Object {
                  "color": "#040404",
                  "marginBottom": 20,
                  "marginTop": 12,
                  "textAlign": "center",
                },
              ]
            }
          >
            Upgrade to
Premium or Unlimited
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "borderColor": "rgba(0,0,0,0.05)",
                "flex": 1,
                "flexDirection": "row",
                "paddingBottom": 20,
                "paddingTop": 20,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 34,
                  },
                  Object {
                    "color": "#0066FF",
                    "marginLeft": 8,
                    "marginRight": 12,
                    "width": 48,
                  },
                  Object {
                    "fontFamily": "FontAwesome5Free-Regular",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontWeight": "400",
                  },
                ]
              }
            >
              
            </Text>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#040404",
                      "fontFamily": "Inter-SemiBold",
                      "fontSize": 17,
                      "lineHeight": 23,
                    },
                    Object {
                      "color": "#040404",
                      "marginBottom": 4,
                    },
                  ]
                }
              >
                Higher Quality voices
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#040404",
                      "fontFamily": "Inter-Regular",
                      "fontSize": 15,
                      "lineHeight": 23,
                    },
                    Object {
                      "color": "#141414",
                    },
                  ]
                }
              >
                Access to our highest quality Premium voices for easier listening. You can preview these Higher Quality voices in the settings screen.
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "borderColor": "rgba(0,0,0,0.05)",
                "flex": 1,
                "flexDirection": "row",
                "paddingBottom": 20,
                "paddingTop": 20,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 34,
                  },
                  Object {
                    "color": "#0066FF",
                    "marginLeft": 8,
                    "marginRight": 12,
                    "width": 48,
                  },
                  Object {
                    "fontFamily": "FontAwesome5Free-Solid",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontWeight": "900",
                  },
                ]
              }
            >
              
            </Text>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#040404",
                      "fontFamily": "Inter-SemiBold",
                      "fontSize": 17,
                      "lineHeight": 23,
                    },
                    Object {
                      "color": "#040404",
                      "marginBottom": 4,
                    },
                  ]
                }
              >
                Voice customization options
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#040404",
                      "fontFamily": "Inter-Regular",
                      "fontSize": 15,
                      "lineHeight": 23,
                    },
                    Object {
                      "color": "#141414",
                    },
                  ]
                }
              >
                Choose between a variety of male and female voices with accents like American, British or Australian English.
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "borderColor": "rgba(0,0,0,0.05)",
                "flex": 1,
                "flexDirection": "row",
                "paddingBottom": 20,
                "paddingTop": 20,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 34,
                  },
                  Object {
                    "color": "#0066FF",
                    "marginLeft": 8,
                    "marginRight": 12,
                    "width": 48,
                  },
                  Object {
                    "fontFamily": "FontAwesome5Free-Regular",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontWeight": "400",
                  },
                ]
              }
            >
              
            </Text>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#040404",
                      "fontFamily": "Inter-SemiBold",
                      "fontSize": 17,
                      "lineHeight": 23,
                    },
                    Object {
                      "color": "#040404",
                      "marginBottom": 4,
                    },
                  ]
                }
              >
                More listening limits
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#040404",
                      "fontFamily": "Inter-Regular",
                      "fontSize": 15,
                      "lineHeight": 23,
                    },
                    Object {
                      "color": "#141414",
                    },
                  ]
                }
              >
                With our Premium and Unlimited subscription you can enjoy higher listening minutes. Unlimited gives you unlimited listening minutes.
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "borderColor": "rgba(0,0,0,0.05)",
                "flex": 1,
                "flexDirection": "row",
                "paddingBottom": 20,
                "paddingTop": 20,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 28,
                  },
                  Object {
                    "color": "#0066FF",
                    "marginLeft": 8,
                    "marginRight": 12,
                    "width": 48,
                  },
                  Object {
                    "fontFamily": "FontAwesome5Free-Solid",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontWeight": "900",
                  },
                ]
              }
            >
              
            </Text>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#040404",
                      "fontFamily": "Inter-SemiBold",
                      "fontSize": 17,
                      "lineHeight": 23,
                    },
                    Object {
                      "color": "#040404",
                      "marginBottom": 4,
                    },
                  ]
                }
              >
                No advertisements
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#040404",
                      "fontFamily": "Inter-Regular",
                      "fontSize": 15,
                      "lineHeight": 23,
                    },
                    Object {
                      "color": "#141414",
                    },
                  ]
                }
              >
                Sponsored content and advertisements helps us continue to provide a free version of Playpost. After upgrading you won’t see any ads, and you’ll be supporting Playpost more directly!
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#040404",
                  "fontFamily": "Inter-Regular",
                  "fontSize": 12,
                  "lineHeight": 15,
                },
                Object {
                  "opacity": 0.5,
                },
              ]
            }
          >
            Payment will be charged to your 
            Apple ID
             account at the confirmation of purchase. Subscription automatically renews unless it is canceled at least 24 hours before the end of the current period. Your account will be charged for renewal within 24 hours prior to the end of the current period. You can manage and cancel your subscriptions by going to your account settings on the App Store after purchase. Refunds are not available for unused portions of a subscription.
          </Text>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 12,
                "marginTop": 12,
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "#040404",
                    "fontFamily": "Inter-Regular",
                    "fontSize": 12,
                    "lineHeight": 15,
                  },
                  Array [
                    Object {
                      "opacity": 0.5,
                    },
                    Object {
                      "color": "#040404",
                      "opacity": 1,
                      "textDecorationLine": "underline",
                    },
                  ],
                ]
              }
            >
              Privacy Policy
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#040404",
                    "fontFamily": "Inter-Regular",
                    "fontSize": 12,
                    "lineHeight": 15,
                  },
                  Object {
                    "opacity": 0.5,
                  },
                ]
              }
            >
               - 
            </Text>
            <Text
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "#040404",
                    "fontFamily": "Inter-Regular",
                    "fontSize": 12,
                    "lineHeight": 15,
                  },
                  Array [
                    Object {
                      "opacity": 0.5,
                    },
                    Object {
                      "color": "#040404",
                      "opacity": 1,
                      "textDecorationLine": "underline",
                    },
                  ],
                ]
              }
            >
              Terms of Use
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#040404",
                    "fontFamily": "Inter-Regular",
                    "fontSize": 12,
                    "lineHeight": 15,
                  },
                  Object {
                    "opacity": 0.5,
                  },
                ]
              }
            >
               - 
            </Text>
            <Text
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "#040404",
                    "fontFamily": "Inter-Regular",
                    "fontSize": 12,
                    "lineHeight": 15,
                  },
                  Array [
                    Object {
                      "opacity": 0.5,
                    },
                    Object {
                      "color": "#040404",
                      "opacity": 1,
                      "textDecorationLine": "underline",
                    },
                  ],
                ]
              }
            >
              Acceptable Use Policy
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 18,
              }
            }
          >
            <View
              style={
                Object {
                  "borderRadius": 6,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#2089dc",
                      "borderColor": "#2089dc",
                      "borderRadius": 6,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "height": 50,
                      "justifyContent": "center",
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontFamily": "Inter-SemiBold",
                        "fontSize": 17,
                        "lineHeight": 23,
                        "paddingBottom": 1,
                        "paddingTop": 2,
                        "textAlign": "center",
                      }
                    }
                  >
                    Already upgraded? Restore purchase
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "marginTop": 18,
                }
              }
            >
              <View
                style={
                  Object {
                    "borderRadius": 6,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "borderColor": "#E9110B",
                        "borderRadius": 3,
                        "borderWidth": 0,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "padding": 8,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#2089dc",
                          "fontFamily": "Inter-SemiBold",
                          "fontSize": 17,
                          "lineHeight": 23,
                          "paddingBottom": 1,
                          "paddingTop": 2,
                          "textAlign": "center",
                        }
                      }
                    >
                      Cancel active subscription?
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
